File: backend/internal/models/meal.go
package models

import "time"

type Meal struct {
	ID int `json:"id"`
	Description string `json:"description"`
	Calories int `json:"calories"`
	Protein float64 `json:"protein"`
	Carbs float64 `json:"carbs"`
	Fat float64 `json:"fat"`
	Timestamp time.Time `json:"timestamp"` 
}


func (m Meal) IsToday() bool {
	now := time.Now()
	return m.Timestamp.Year() == now.Year() &&
	m.Timestamp.Month() == now.Month() &&
	m.Timestamp.Day() == now.Day()
}

File: mobile/lib/models/meal.dart
class Meal {
	final int id;
	final String description;
	final int calories;
	final double protein;
	final double carbs;
	final double fat;
	final DateTime timestamp;
	

	Meal({
		required this.id,
		required this.description,
		required this.calories,
		required this.protein,
		required this.carbs,
		required this.fat,
		required this.timestamp,
	});

	factory Meal.fromJson(Map<String, dynamic> json) {
		return Meal(
			id: json['id'] ?? 0,
			description: json['description'],
			calories: json['calories'],
			protein: (json['protein'] as num).toDouble(),
			carbs: (json['carbs'] as num).toDouble(),
			fat: (json['fat'] as num).toDouble(),
			timestamp: DateTime.parse(json['timestamp']),
		);
	}



}
